dc/core/PixelStreamUpdater.cpp

```diff
@@ -75,21 +83,10 @@ QImage PixelStreamUpdater::getTileImage( const uint frameIndex,
     // to determine which frames can be deleted from the _frames map
     _requestedFrameIndex = std::max( _requestedFrameIndex, frameIndex );
 
-    // find the segments for the requested frame
-    deflect::FramePtr frame;
-    for( const auto& i : _frames )
-    {
-        if( i.first == frameIndex )
-        {
-            frame = i.second;
-            break;
-        }
-    }
-
-    if( !frame )
+    if( _frames.empty( ))
         throw std::runtime_error( "No frames yet" );
 
-    const auto& segment = frame->segments.at( _visibleSet[tileIndex] );
+    const auto& segment = _currentFrame()->segments.at( _visibleSet[tileIndex]);
```

Comment:
Since the image requests have been made synchronous, tiles are always requested for the current frame, at the moment the tilesChanged() or sourceParamsChanged() signals are emitted. There are no pending requests for previous tiles. I removed the _frames and _frameIndex from PixelStreamUpdater, and carefully checked the behaviour again. Everything behaves as expected, bugs are fixed, and I observed no differences in performance.